Core/src/ApiHelperTrait.php

```diff
@@ -264,4 +265,45 @@ private function splitOptionalArgs(array $input, array $extraAllowedKeys = []):
 
         return [$input, $optionalArgs];
     }
+
+    /**
+     * Helper method used to validate optons based on the supplied $optionTypes
+     * $optionTypes can be an array of string keys, a protobuf Message classname, or a
+     * the CallOptions classname. Parameters are split and returned in the order
+     * that the options types are provided.
+     */
+    private function validateOptions(array $options, array|Message|string ...$optionTypes): array
+    {
+        $splitOptions = [];
+        foreach ($optionTypes as $optionType) {
+            if (is_array($optionType)) {
```

Comment:
I am guessing here, but basically we want to do this:
```
$options = [
    'cacheItemPool' => $cacheInstance
    'authHandler' => $authHandler
]
$validations = [
    cacheItemPoolInterface::class,
    authHandler::class
]
$this->validateOptions($options, $validations);
```

Am I correct? If so, why do we take an array on the option type? Is it for the case of an option taking `null|itemCachePoolInterface`?